libavformat/movenccenc.c

```diff
@@ -640,5 +640,12 @@ void ff_mov_cenc_free(MOVMuxCencContext* ctx)
     if (ctx->encryption_scheme == MOV_ENC_CENC_AES_CBC_PATTERN && ctx->aes_cbc) {
         av_freep(&ctx->aes_cbc);
     }
-    av_freep(&ctx->auxiliary_info);
+    if (ctx->auxiliary_info_size > 0) {
+        av_freep(&ctx->auxiliary_info);
+    }
+    if (ctx->auxiliary_info_sizes_alloc_size > 0) {
```

Comment:
- The same question here: do we need to check ctx->auxiliary_info_sizes_alloc_size? How does it help?
- Needs some comments and explain why to do this check.

Reply:
Those checks may yield incorrect behavior as av_freep has a side effect of nulling the actual pointer as it takes a **, so by not calling av_freep the pointer itself may never be nulled.